### ISSUE TITLE
Image Classifier

### ISSUE THREAD

**Issue by Seenamtl** [Seenamtl/uni#13]
`import tensorflow as tf
from tensorflow.keras.applications import MobileNetV2
from tensorflow.keras.applications.mobilenet_v2 import preprocess_input, decode_predictions
from tensorflow.keras.preprocessing.image import load_img, img_to_array
import numpy as np

def load_and_prepare_image(image_path, target_size=(224, 224)):
    """
    Loads and preprocesses an image for the model.
    
    Args:
        image_path (str): Path to the image file.
        target_size (tuple): Desired size for the image.
        
    Returns:
        np.array: Preprocessed image ready for classification.
    """
    image = load_img(image_path, target_size=target_size)
    image_array = img_to_array(image)
    image_array = np.expand_dims(image_array, axis=0)
    return preprocess_input(image_array)

def classify_image(model, image_path):
    """
    Classifies the input image using the given model.
    
    Args:
        model: Pre-trained TensorFlow/Keras model.
        image_path (str): Path to the image file.
    
    Returns:
        list: Decoded predictions with class labels and probabilities.
    """
    prepared_image = load_and_prepare_image(image_path)
    predictions = model.predict(prepared_image)
    return decode_predictions(predictions, top=3)[0]

def main():
    print("Image Classifier Tool")
    print("Type 'exit' to quit the program.")
    
    # Load the pre-trained model
    print("Loading model...")
    model = MobileNetV2(weights='imagenet')
    print("Model loaded successfully.")
    
    while True:
        image_path = input("\nEnter the path to an image file: ")
        
        if image_path.lower() == 'exit':
            print("Exiting the tool. Goodbye!")
            break
        
        try:
            print("\nClassifying image...")
            predictions = classify_image(model, image_path)
            
            print("Top Predictions:")
            for i, (imagenet_id, label, prob) in enumerate(predictions):
                print(f"{i + 1}. {label} ({prob:.2f}%)")
        
        except Exception as e:
            print(f"Error: {e}")
            print("Please check the file path or try again.")

if __name__ == "__main__":
    main()
`
